web/src/message_view.js

```diff
@@ -87,7 +87,17 @@ export function changehash(newhash, trigger) {
         return;
     }
     message_viewport.stop_auto_scrolling();
-    browser_history.set_hash(newhash);
+
+    // It is important to use `replaceState` rather than `replace` in case
+    // of "retarget topic location" because we retarget to the last unread.
+    // Using `replce` would update the hash and load it once more.
+    // This would impact the `narrow_pointer` for users with
+    // `mark_read_on_scroll_state_policy` set.
+    if (trigger === "retarget topic location") {
+        window.history.replaceState(null, "", newhash);
+    } else {
+        browser_history.set_hash(newhash);
+    }
```

Comment:
I added this edge case, without which the back operation won't work.

Reply:
Can you explain scenario in a bit more detail?

Reply:
I assume part of the operative but here is that first parameter being `null`; we store metadata for the `back` button in `update_current_history_state_data`; is it correct to set null here? It might be, but we should explain that explicitly in a comment here. I made an attempt at this but I'd appreciate your review on whether the language is correct.

``` diff
diff --git a/web/src/message_view.js b/web/src/message_view.js
index e3a1e86503..3f0c0eed3f 100644
--- a/web/src/message_view.js
+++ b/web/src/message_view.js
@@ -88,13 +88,16 @@ export function changehash(newhash, trigger) {
     }
     message_viewport.stop_auto_scrolling();
 
-    // It is important to use `replaceState` rather than `replace` in case
-    // of "retarget topic location" because we retarget to the last unread.
-    // Using `replce` would update the hash and load it once more.
-    // This would impact the `narrow_pointer` for users with
-    // `mark_read_on_scroll_state_policy` set.
     if (trigger === "retarget topic location") {
-        window.history.replaceState(null, "", newhash);
+        // It is important to use `replaceState` rather than `replace`
+        // here for the `back` button to work; we don't want to use
+        // any metadata potentially stored by
+        // update_current_history_state_data associated with an old
+        // URL for the target conversation, and conceptually we want
+        // to replace the inaccurate/old URL for the conversation with
+        // the current/corrected value.
+        const current_state = window.history.state;
+        window.history.replaceState(current_state, "", newhash);
     } else {
         browser_history.set_hash(newhash);
     }
```

Is that correct? I am kinda guessing your intent here, so it's very possible this is nonsense/wrong.

Reply:
In particular, I don't quite understand why it might be possible that there was state on the previous URL that we need to manage.

Maybe best if you start a #frontend thread about this detail to discuss further.

Reply:
Did this frontend thread happen? We should add a cross-link here if so.

Reply:
ya it happened, let me look it up